include/rocksdb/slice.h

```diff
@@ -212,6 +212,8 @@ class PinnableSlice : public Slice, public Cleanable {
 
   inline bool IsPinned() const { return pinned_; }
 
+  inline void Pin() { pinned_ = true; }
```

Comment:
I am concerned that this API is hard to use correctly.

Reply:
I didn't find a better way to set `pinned_`

Reply:
Maybe we don't need to set pinned_ to true in here.